### PR TITLE
Use grpc protocol for remote cache on Mac

### PR DESCRIPTION
**Pull request by meteorcloudy**

### CODE REVIEW

File: buildkite/bazelci.py [hunk 1]
```diff
@@ -1898,7 +1898,7 @@ def remote_caching_flags(platform, accept_cached=True):
 
     if is_mac() and is_lab_machine():
         # Use a local cache server for our physical macOS machines in the lab.
-        flags = ["--remote_cache=http://[2a00:79e1:abc:8602:a28c:fdff:fed0:ec39]"]
+        flags = ["--remote_cache=grpc://[2a00:79e1:abc:8602:a28c:fdff:fed0:ec39]"]
```
Comment (coeuvre) [inline buildkite/bazelci.py:1901]: Can you check which port the grpc endpoint is bound to (and make sure it's accessible from the CI worker)? 

For reference: https://github.com/buchgr/bazel-remote?tab=readme-ov-file#command-line-flags

[the remark this answers is not in the document]
Comment (coeuvre) [reply buildkite/bazelci.py:1901]: grpcs requires setting up certificate. since we were using http, not https, it should be fine to use grpc.

Comment (meteorcloudy) [reply buildkite/bazelci.py:1901]: It should be the default 9092 port and is accessible.

[the remark this answers is not in the document]
Comment (meteorcloudy) [reply buildkite/bazelci.py]: SGTM

**Review by coeuvre — APPROVED** [review]

**Pull request merged**